platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -1165,4 +1171,326 @@ private interface AddressBookCallback {
         void apply(int index, NodeId nodeId, Address address, String nodeAlias)
                 throws KeyStoreException, KeyLoadingException;
     }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////
+    //////////////////////////////////// MIGRATION METHODS //////////////////////////////////////////
+    /////////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Performs any necessary migration steps to ensure the key storage is up-to-date.
+     * <p>
+     * As of release 0.56 the on-disk cryptography should reflect the following structure:
+     * <ul>
+     *     <li>s-private-alias.pem - the private signing key </li>
+     *     <li>s-public-alias.pem - the public signing certificates of each node</li>
+     *     <li>all *.pfx files moved to <b>OLD_PFX_KEYS</b> sub-directory and no longer used.</li>
+     *     <li>all agreement key material is deleted from disk.</li>
+     * </ul>
+     *
+     * @return this {@link EnhancedKeyStoreLoader} instance.
+     */
+    @NonNull
+    public EnhancedKeyStoreLoader migrate() throws KeyLoadingException, KeyStoreException {
+        logger.debug(STARTUP.getMarker(), "Starting key store migration");
+        final Map<NodeId, PrivateKey> pfxPrivateKeys = new HashMap<>();
+        final Map<NodeId, Certificate> pfxCertificates = new HashMap<>();
+
+        // delete agreement keys
+        deleteAgreementKeys();
+
+        // create PEM files for signing keys and certs.
+        long errorCount = extractPrivateKeysAndCertsFromPfxFiles(pfxPrivateKeys, pfxCertificates);
+
+        if (errorCount == 0) {
+            // validate only when there are no errors extracting pem files.
+            errorCount = validateKeysAndCertsAreLoadableFromPemFiles(pfxPrivateKeys, pfxCertificates);
+        }
+
+        if (errorCount > 0) {
+            // roll back due to errors.
+            // this deletes any pem files created, but leaves the agreement keys deleted.
+            logger.debug(STARTUP.getMarker(), "Due to {} errors, reverting pem file creation.", errorCount);
+            rollBackSigningKeysAndCertsChanges(pfxPrivateKeys, pfxCertificates);
+        } else {
+            // cleanup pfx files by moving them to sub-directory
+            cleanupMovePfxFilesToSubDirectory();
+        }
+
+        logger.trace(STARTUP.getMarker(), "Completed migration of key store.");
```

Comment:
I bumped this and the starting log up to info since there is a info message related to migration that would be out of context without at least the starting log message.